splink/settings.py

```diff
@@ -518,6 +518,6 @@ def human_readable_description(self):
     @property
     def salting_required(self):
         for br in self._blocking_rules_to_generate_predictions:
-            if br.salting_partitions > 1:
+            if isinstance(br, SaltedBlockingRule):
```

Comment:
Does this also need to check the sql dialect?

Reply:
Well spotted - I've added the check to the SaltedBlockingRule class in https://github.com/moj-analytical-services/splink/pull/1701/commits/b49bb06cf0564f19eb38b09fb5b78a3233a86c01